### DISCUSSION TITLE
Can knots store metadata - YarnEditor as an alternative to Inkle (for mobile OS)

### DISCUSSION THREAD

**Discussion by blurymind** [inkle/ink#719] [Show and tell]
Hi everyone,  I am the developer of YarnEditor and I just added ink file support to it
https://yarnspinnertool.github.io/YarnEditor/

in the editor I am allowing the user to organise knots as nodes, search between nodes and see their connection
The ink files load/parse and can be previewed, but there is one tiny problem atm - ink knots need to somehow save x and y values, so when the user moves them around, their positions can be stored without affecting the game.

Is there a way in ink syntax to give knots extra metadata?
What would be the best way in your opinion? This data needs to not show in the text editor part, as the user is not allowed to type their x/y position - that gets assigned by them simply being dragged around.

![inkPlaytest](https://user-images.githubusercontent.com/6495061/139660942-2d80b8e0-245b-49ab-bc32-ef5d3e3f883d.gif)

I saw that ink devs are not the biggest fans of the node tree approach and still want to later on add support for a single text file way, but for now I want to give this a try and see how it works. 

Nodes have one big advantage in yarn and that is they allow you to break down your narrative into chunks of text, rather than have one giant text file to scroll through. You can search data by knot name or knot contents/tags. Heres how it looks like with a yarn file
![yarnImproveSearch](https://user-images.githubusercontent.com/6495061/139661076-27e17280-1e12-45dc-9ed1-534a321ad92c.gif)

**Comment by joningold** [comment]
Maybe add x/y as a comment after the name of the ink knot and parse it from
there? It’d be relatively unobtrusive in the ink files.

**Reply by blurymind** [reply]
Like this?

```
-> knot

=== knot
//metadata: x:22, y:200
Once upon a time...

 * There were two choices.
    Poarot: how dare you
    -> anotherKnot
 * There were four lines of content.
    Dont steal my thunder
    ** Go to another?
        No way!
    ** End it here
        Yes way
    -- I am continuing blah blah
    And blah blan

- They lived happily ever after
    -> END
    
=== anotherKnot
//metadata: x:22, y:200
Test me
-> END

```

The trouble with that is that I need to prevent the user from seeing the comment when they open to edit a knot. I guess I could give it a try

**Comment by blurymind** [comment]
I actually prefer to put the comment on the same line as the knot title like this:
```
-> knot

=== knot === // yarn-editor-metadata: {x: 12, y: 22}
Once upon a time...

 * There were two choices.
    Poarot: how dare you
    -> anotherKnot
 * There were four lines of content.
    Dont steal my thunder
    ** Go to another?
        No way!
    ** End it here
        Yes way
    -- I am continuing blah blah
    And blah blan

- They lived happily ever after
    -> END
    
=== anotherKnot === // yarn-editor-metadata: {x: 12, y: 22}
Test me
-> END
```

will that be ok?
